src/lib/wallet/GoodWalletClass.js

```diff
@@ -359,7 +359,11 @@ export class GoodWallet {
 
   async getNextClaimTime(): Promise<any> {
```

Comment:
Have you checked all places where those methods used ? are they wrapped in try catch and handles exception correctly?